### PR TITLE
Ability to use IndexTriangles

### PR DESCRIPTION
**Pull request by nacmartin**
This provides the ability to specify indices. There is a nice explanation of this technique here http://www.opengl-tutorial.org/intermediate-tutorials/tutorial-9-vbo-indexing/

Here is a gist that uses this feature: https://gist.github.com/nacmartin/fb79c2a0a0b26d9bd2c233f8f2da7564 

Note that this is a change in the API. It doesn't break compatibility, as it only provides a new type for a union type, but I will understand if it is not merged.

This PR addresses #37

### CODE REVIEW

**Comment by nacmartin** [comment]
I just noticed rendering more complex meshes that the vertexes (not the indices) need to be reversed. Fixed in next commit.

File: src/Native/WebGL.js [hunk 1]
```diff
@@ -203,7 +205,7 @@ var _elm_community$elm_webgl$Native_WebGL = function () {
     A2(_elm_lang$core$List$map, function (elem) {
       dataFill(array, attributeInfo.size, data_idx, elem, attribute.name);
       data_idx += attributeInfo.size * elem_length;
-    }, bufferElems);
+    }, _elm_lang$core$List$reverse(bufferElems));
```
Comment (w0rm) [inline src/Native/WebGL.js:208]: @nacmartin I'm curious about this particular change. 
- Won't it affect the other cases?
- Could we use `foldr` to not iterate over the list twice?

**Comment by w0rm** [comment]
@nacmartin it also looks that `doBindSetup` and `doBindSetupIndexed` are almost identical, maybe we can have one function that takes indices.

**Comment by w0rm** [comment]
Another idea would be to add an `Either` of each case, indexed and non-indexed. This will probably change the way of creating drawables, but this change is major anyway.

File: src/Native/WebGL.js [hunk 2]
```diff
@@ -343,7 +373,13 @@ var _elm_community$elm_webgl$Native_WebGL = function () {
         gl.enableVertexAttribArray(attribLocation);
 
         if (buffer.buffers[attribute.name] === undefined) {
-          buffer.buffers[attribute.name] = do_bind_attribute(gl, attribute, render.buffer._0, renderType.elemSize);
+          var bufferElems;
+          if (renderType.indexed) {
+            bufferElems = render.buffer._0._0, renderType.elemSize;
```
Comment (w0rm) [inline src/Native/WebGL.js:378]: This looks wrong to me, you are using comma operator here

**Comment by w0rm** [comment]
I've rebased this pull request, so we can continue discussing it here: https://github.com/elm-community/webgl/pull/8

**Pull request closed without merging**